core-bundle/src/String/SimpleTokenParser.php

```diff
@@ -118,7 +118,7 @@ private function replaceTokens(string $subject, array $data): string
     {
         // Replace tokens
         return preg_replace_callback(
-            '/##([^=!<>\s]+?)##/',
+            '/##([^#=!<>\s]+?)##/',
```

Comment:
```suggestion
            '/##([^#=!<>\s][^=!<>\s]*?)##/',
```
Only disallow the `#` at the beginning?